components/bl-cropper-component/src/cropper-components.js

```diff
@@ -0,0 +1,142 @@
+  );
+}
+
+export function HeaderToolbar({ component, cropperRef, onSave, image, setImage }) {
+  const { addImageButtonVisibility, cleanButtonVisibility, saveButtonVisibility, downloadButtonVisibility } = component;
+  const visibility = (
+    addImageButtonVisibility || cleanButtonVisibility || saveButtonVisibility || downloadButtonVisibility
+  );
+
+  const clean = () => setImage('');
+  const uploadImage = e => upload(e, setImage);
+
+  const saveImage = () => {
+    const croppedCanvasURL = cropperRef.current.getCroppedCanvas().toDataURL();
+
+    dataURLToBlob(croppedCanvasURL).then(fileContent => onSave({ fileContent }));
+  };
+
+  const downloadImage = () => {
+    const croppedCanvasURL = cropperRef.current.getCroppedCanvas().toDataURL();
+
+    download(croppedCanvasURL, 'cropped.png');
+  };
+
+  if (!image || !visibility) {
+    return null;
+  }
+
+  return (
+    <div className="header-toolbar">
+      <div className="buttons-wrapper">
+        <ImageUploadButton onChange={ uploadImage } label="Add image" visibility={ addImageButtonVisibility }/>
+        <Button onClick={ clean } label="Clean" visibility={ cleanButtonVisibility }/>
+      </div>
+      <div className="buttons-wrapper">
+        <Button onClick={ saveImage } label="Save" visibility={ saveButtonVisibility }/>
+        <Button onClick={ downloadImage } label="Download" visibility={ downloadButtonVisibility }/>
+      </div>
+    </div>
+  );
+}
+
+export function FooterToolbar({ cropperRef, image, toolbarVisibility }) {
+  const toolbar = useMemo(() => {
+    const zoomIn = () => cropperRef.current.zoom(0.1);
+    const zoomOut = () => cropperRef.current.zoom(-0.1);
+    const rotateLeft = () => cropperRef.current.rotate(-45);
+    const rotateRight = () => cropperRef.current.rotate(45);
+    const scaleX = () => {
+      const newScaleX = -cropperRef.current.getImageData().scaleX;
+
+      cropperRef.current.scaleX(newScaleX || -1);
+    };
+    const scaleY = () => {
+      const newScaleY = -cropperRef.current.getImageData().scaleY;
+
+      cropperRef.current.scaleY(newScaleY || -1);
+    };
+
+    return [
+      [{ label: 'zoom_in', onClick: zoomIn }, { label: 'zoom_out', onClick: zoomOut }],
+      [{ label: 'rotate_left', onClick: rotateLeft }, { label: 'rotate_right', onClick: rotateRight }],
+      [{ label: 'swap_horiz', onClick: scaleX }, { label: 'swap_vert', onClick: scaleY }],
+    ];
+  }, [cropperRef]);
+
+  if (!toolbarVisibility || !image) {
+    return null;
+  }
+
+  return (
+    <div className="footer-toolbar">
+      { toolbar.map((buttons, index) => (
+        <ButtonsGroup buttons={ buttons } key={ index }/>
+      )) }
+    </div>
+  );
+}
+
+function ButtonsGroup({ buttons }) {
+  return (
+    <div className="buttons-group">
+      { buttons.map((button, index) => (
+        <Button
+          className="material-icons-round"
+          onClick={ button.onClick }
+          label={ button.label }
+          visibility={ true }
```

Comment:
`visibility={ true }` is it correct?

Reply:
Yes, that is correct. 
The `visibility `attribute on the `Button` component is needed in `HeaderToolbar`, where I also use this component. In `ButtonsGroup` it should be `true`.